src/schedule/zephyr_domain.c

```diff
@@ -55,6 +56,14 @@ struct zephyr_domain {
 /* perf measurement windows size 2^x */
 #define CYCLES_WINDOW_SIZE	10
 
+#if CONFIG_WDT_DW
+#define GET_DEVICE_LIST(node) DEVICE_DT_GET(node),
+
+const struct device *watchdog[CONFIG_MAX_CORE_COUNT] = {
```

Comment:
This needs to be in uncache section and aligned for cache line/size - can you confirm.

Reply:
It is an array of pointers to the struct device. Its constant and don't think it need any special handling.

Reply:
This depends on what other data is sharing the same cache line. If its a one time write, then making sure nothing else is on the same cache line is still needed

Reply:
@softwarecki as defined here it's an array of constant pointers, not a constant array. I.e. you cannot use pointers in this array to modify devices that it points to, but you can well modify contents of this array. The array itself isn't constant. If you want to make the array constant instead you need
```
static struct device * const watchdog[CONFIG_MAX_CORE_COUNT] = {
```
(btw, yes, you want this array `static`). If you want both - constant array of constant pointers you need
```
static const struct device * const watchdog[CONFIG_MAX_CORE_COUNT] = {
```
Now as is the array isn't constant, so it ends up in .bss, and as such it is uncached, it shares cache lines in the same section with other uncached objects, so cache is never used for them and can never go out of sync. If you make the array itself constant, it will go to .rodata, and that will be cached. But then it will be sharing cache lines with other constant data and as such that data is never modified, so again, no loss of synchronisation is possible. So, @lgirdwood I don't think we need any special handling here. If we deed, we'd need it in many other locations throughout the code and things would be broken right now.

Reply:
I'm fine if the data is all const, but my understanding here is that this is NOT the case ? and that ```CONFIG_MAX_CORE_COUNT``` means this data will be shared across cores (and we need to spell out this data should be in uncache and never directly accessed via cache (unless we have an access mechanism)). Can someone confirm usage ?